src/mrack/providers/openstack.py

```diff
@@ -278,6 +278,24 @@ def validate_host(self, req):
 
         return True
 
+    async def prepare_provisioning(self, reqs):
+        """
+        Prepare provisioning.
+
+        Load missing images if they are not in provisioning-config.yaml
+        """
+        prepare_images = []
+
+        for req in reqs:
+            req_img = req["image"]
```

Comment:
What about:
```
        prepare_images = unique([req["image"] for req in req if req["image"] not in self.images])
        if prepare_images:
             im_list = ', '.join(prepare_images)
             logger.debug(f"{self.dsp_name}: Loading image info for: '{im_list}'")
             await self.load_images(prepare_images)
             logger.debug(f"{self.dsp_name}: Loading images info done.")
```

That way it will print only unique info.